File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int curr = 0;
        HashMap<Integer,Integer> map = new HashMap<>();
        for(int i=0;i<nums.length;i++)
        {
            curr = nums[i];
            if(map.containsKey(target-curr)){
                return new int[]{i, map.get(target-curr)};
            }
            map.put(nums[i],i);
        }
        return new int[]{};
    }
}

File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>  map;
        int curr = 0;
        for(int i=0;i<nums.size();i++){
            curr = nums[i];
            if(map.find(target-curr) != map.end()){
                return {map[target-curr],i};
            }
            map[nums[i]] = i;
        }
        return {};
    }
};